src/LinuxTracing/PerfEvent.h

```diff
@@ -359,6 +363,8 @@ struct SchedSwitchWithStackPerfEventData {
   // UprobesReturnAddressManager::PatchSample.
   [[nodiscard]] uint8_t* GetMutableStackData() const { return data.get(); }
   [[nodiscard]] uint64_t GetStackSize() const { return dyn_size; }
+  [[nodiscard]] pid_t GetCallstackPid() const { return prev_pid_or_minus_one; }
```

Comment:
In unwinding it is only used to possibly read the process memory. If we would receive a SchedSwitch**WithStack**PerfEvent where there prev_pid is minus one (what I am actually not sure would ever happen, as I would assume we don't have a callstack here), then unwinding will still be possible, unless process memory is required.

I also checked the code in libunwindstack. Creating process memory for pid -1 is fine. We would fail when attempting to read the memory, which would result in an unwinding error.

Reply:
This happens when the switch out is caused by the process exiting. There are a couple of comments around, e.g., in `TracerImpl`:
```
                // As the tracepoint data does not include the pid of the process that the thread
                // being switched out belongs to, we use the pid set by perf_event_open in the
                // corresponding generic field of the PERF_RECORD_SAMPLE.
                // Note, though, that this value is -1 when the switch out is caused by the thread
                // exiting. This is not the case for data.prev_pid, whose value is always correct as
                // it comes directly from the tracepoint data.
```
I don't know if a callstack is present in that case. We might want to check. If yes, it probably contains the call to exit system call.

I think we might want to call this method `GetCallstackPidOrMinusOne`, to signal the possibility of the -1, just like `prev_pid_or_minus_one`, even if the -1 is not really a problem where the method is used.

Reply:
Edit: process exiting -> thread exiting

Reply:
>I think we might want to call this method GetCallstackPidOrMinusOne, to signal the possibility of the -1, just like prev_pid_or_minus_one, even if the -1 is not really a problem where the method is used.

The problem is, that we introduced this method for the templated `Unwind` method, such that we can simply access the pid here, and get the one that belongs to the callstack. For sure we could make the callstack pid a parameter, though.

But I don't see the problem yet. Shall we discuss offline?

Reply:
Yeah, my idea was to rename all the `GetCallstackPid`. It's like a virtual method, most implementations would never return -1, but some could, and we consider that in the name.

Still, let me know if you'd like to discuss online, maybe I'm missing something else.

Reply:
Okay, renamed all the functions to `GetCallstackPidOrMinusOne`.